Step04/app/src/main/java/com/kwoolytech/step04/MainActivity.java

```diff
@@ -34,34 +35,35 @@ protected void onCreate(Bundle savedInstanceState) {
         ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(this, drawer, toolbar,
                 R.string.navigation_drawer_open, R.string.navigation_drawer_close);
 
-        drawer.setDrawerListener(toggle);
         toggle.syncState();
 
         NavigationView navigationView = (NavigationView)findViewById(R.id.nav_view);
         navigationView.setNavigationItemSelectedListener(this);
 
-        dataModel = new HansWeatherDataModel();
+        dataModel   = new HansWeatherDataModel();
         httpClient  = new KwoolyHttp(MainActivity.this);
 
         initializeUi();
         initializeKwoolyHttpCallbackFunction();
 
-        queryWeatherJsonData();
+        queryWeatherJsonData(37.49, 127.01);
 
         return;
     }
 
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if (requestCode == CommonTool.CODE_PROCESS_MAP_LOCATION_REQUEST) {
+            if (resultCode == RESULT_OK) {
+                queryWeatherJsonData(data.getDoubleExtra("Lat", 0), data.getDoubleExtra("Lng", 0));
+            }
+        }
```

Comment:
@kwoolytech let's show toast here indicating no location picking from the location picker screen inside else { } statement 

https://developer.android.com/guide/topics/ui/notifiers/toasts.html